src/Form/Form.tsx

```diff
@@ -1,20 +1,74 @@
 import * as React from "react";
 import * as cx from "classnames";
-import { props, t } from "../utils";
+import { StatefulButton } from "../Button/StatefulButton";
+import View from "react-flexview";
 
 export namespace Form {
-  export type Props = React.HTMLProps<HTMLFormElement> & {
-    ref: (ref: Form | null) => void;
+  export type Props = View.Props & {
+    onSubmit: () => Promise<void>;
+    submitLabel: string;
+    submitProcessingLabel?: string;
+    renderSubmit?: (buttonProps: StatefulButton.Props) => JSX.Element;
   };
 }
 
-export const Props = t.Object;
-
-@props(Props)
 export class Form extends React.PureComponent<Form.Props> {
+  formRef = React.createRef<HTMLFormElement>();
+
+  onFormSubmit: React.ReactEventHandler<HTMLFormElement> = e => {
+    if (this.formRef.current) {
```

Comment:
this code is temporary: `Form` should actually enforce the error guidelines in the future, such as avoid stateful buttons; it will thus optionally show an error and track the Promise state internally